File: src/main/scala/io/github/jbellis/brokk/analyzer/CallSite.scala
package io.github.jbellis.brokk.analyzer

/**
 * Represents a call site within source code
 *
 * @param signature The fully-qualified method signature
 * @param sourceLine The exact line of code where the call is made (trimmed of whitespace)
 */
case class CallSite(
  signature: String,
  sourceLine: String
)

File: src/main/java/io/github/jbellis/brokk/analyzer/CallSite.java
package io.github.jbellis.brokk.analyzer;

/**
 * A record representing a call site in source code.
 *
 * @param target The CodeUnit of the method being called/calling
 * @param sourceLine The actual source line where the call occurs
 */
public record CallSite(CodeUnit target, String sourceLine) implements Comparable<CallSite> {
    @Override
    public int compareTo(CallSite other) {
        return target.identifier().compareTo(other.target.identifier());
    }
}
